parts/configure-swarmmode-cluster.sh

```diff
@@ -0,0 +1,302 @@
+#!/bin/bash
+
+###########################################################
+# Configure Swarm Mode One Box
+#
+# This installs the following components
+# - Docker
+# - Docker Compose
+# - Swarm Mode masters
+# - Swarm Mode agents
+###########################################################
+
+set -x
+
+echo "starting Swarm Mode cluster configuration"
+date
+ps ax
+
+DOCKER_COMPOSE_VERSION="1.9.0"
+#############
+# Parameters
+#############
+
+MASTERCOUNT=${1}
+MASTERPREFIX=${2}
+MASTERFIRSTADDR=${3}
+AZUREUSER=${4}
+POSTINSTALLSCRIPTURI=${5}
+BASESUBNET=${6}
+VMNAME=`hostname`
+VMNUMBER=`echo $VMNAME | sed 's/.*[^0-9]\([0-9]\+\)*$/\1/'`
+VMPREFIX=`echo $VMNAME | sed 's/\(.*[^0-9]\)*[0-9]\+$/\1/'`
+
+echo "Master Count: $MASTERCOUNT"
+echo "Master Prefix: $MASTERPREFIX"
+echo "Master First Addr: $MASTERFIRSTADDR"
+echo "vmname: $VMNAME"
+echo "VMNUMBER: $VMNUMBER, VMPREFIX: $VMPREFIX"
+echo "BASESUBNET: $BASESUBNET"
+echo "AZUREUSER: $AZUREUSER"
+
+###################
+# Common Functions
+###################
+
+ensureAzureNetwork()
+{
+  # ensure the host name is resolvable
+  hostResolveHealthy=1

```

Comment:
We discovered yesterday, we do not depend on iDNS - the VM names, and lines 49-65 can be removed here since that will unblock the Bring your own DNS scenario.  The remaining network checks are necessary to ensure a healthy network.